### PR TITLE
Add ingest-attachment support for per document `indexed_chars` limit

### LINKED ISSUE #28942: ingest-attachment support for per document `indexed_chars` limit
Coming from this discussion: https://discuss.elastic.co/t/how-to-control-the-indexed-chars-value-on-a-ingest-attachment-pipeline/123073/4

We today support a global `indexed_chars` processor parameter. But in some cases, users would like to set this limit depending on the document itself.
It used to be supported in mapper-attachments plugin by extracting the limit value from a meta field in the document sent to indexation process.

Here is my proposal.
We should add an option like reading this limit value from the document itself by adding a setting like `indexed_chars_field`.

Then we could do something:

```
PUT _ingest/pipeline/attachment
{
  "description" : "Extract attachment information. Used to parse pdf and office files",
  "processors" : [
    {
      "attachment" : {
        "field" : "data",
        "indexed_chars_field" : "size"
      }
    }
  ]
}
```
 

Then index either:

```
PUT index/doc/1?pipeline=attachment
{
  "data": "BASE64"
}
```

Which will use the default value (or the one defined by `indexed_chars`)

Or

```
PUT index/doc/2?pipeline=attachment
{
  "data": "BASE64",
  "size": 1000 
}
```

I'll propose hopefully soon a PR for it unless someone in the meantime reject that feature request or propose another implementation for it.

### PR DESCRIPTION
**Pull request by dadoonet**
We today support a global `indexed_chars` processor parameter. But in some cases, users would like to set this limit depending on the document itself.
It used to be supported in mapper-attachments plugin by extracting the limit value from a meta field in the document sent to indexation process.

We add an option which reads this limit value from the document itself
by adding a setting named `indexed_chars_field`.

Which allows running:

```
PUT _ingest/pipeline/attachment
{
  "description" : "Extract attachment information. Used to parse pdf and office files",
  "processors" : [
    {
      "attachment" : {
        "field" : "data",
        "indexed_chars_field" : "size"
      }
    }
  ]
}
```

Then index either:

```
PUT index/doc/1?pipeline=attachment
{
  "data": "BASE64"
}
```

Which will use the default value (or the one defined by `indexed_chars`)

Or

```
PUT index/doc/2?pipeline=attachment
{
  "data": "BASE64",
  "size": 1000
}
```

Closes #28942.

### CODE REVIEW

File: docs/plugins/ingest-attachment.asciidoc [hunk 1]
```diff
@@ -99,6 +100,101 @@ NOTE: Extracting contents from binary data is a resource intensive operation and
...
+  "found": true,
+  "_index": "my_index",
+  "_type": "_doc",
+  "_id": "my_id",
+  "_version": 1,
+  "_source": {
+    "data": "e1xydGYxXGFuc2kNCkxvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0DQpccGFyIH0=",
+    "attachment": {
+      "content_type": "application/rtf",
+      "language": "ro",
```
Comment (martijnvg) [inline docs/plugins/ingest-attachment.asciidoc]: The doc build fails, because `sl` is returned instead.

**Review by martijnvg — APPROVED** [review]
Thanks @dadoonet!

LGTM, assuming the pr build passes.

File: docs/plugins/ingest-attachment.asciidoc (hunk 1 above)
Comment (dadoonet) [reply docs/plugins/ingest-attachment.asciidoc]: True. TBH I was expecting that as I did not run the doc check myself but let the CI do it (I was in a taxi when I pushed my PR ;) )... 
I'll fix it, run a check locally and let the CI say ok. 
Thanks!

**Comment by dadoonet** [comment]
@elasticmachine restest please

**Comment by dadoonet** [comment]
@elasticmachine test this please

**Comment by dadoonet** [comment]
@martijnvg Apparently CI is still unhappy but I'm unsure if it's my fault. Could you check it please?

**Comment by martijnvg** [comment]
@dadoonet The test failure looks unrelated to this change. Maybe rebase master and re-run the pr build?

**Comment by dadoonet** [comment]
Hmmm. Sounds like I forgot to backport this PR in 6.3 branch.
I just removed the `6.3.0` label then.
May be I can still backport to 6.x (6.4.0).

**Pull request merged**